pkg/sql/catalog/randgen/api.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package randgen
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen/randgencfg"
+	"github.com/cockroachdb/cockroach/pkg/util/randident"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+)
+
+// TestSchemaGeneratorConfig is an alias for randgencfg.Config.
+type TestSchemaGeneratorConfig = randgencfg.Config
+
+// TestSchemaGenerator can be used to generate random SQL objects.
+// See NewTestSchemaGenerator().
+//
+// A design goal for this facility is to ensure that the generation is
+// fast in the common case, so it can be used to generate large
+// schemas quickly.
+type TestSchemaGenerator interface {
```

Comment:
Should this be named `RandomSchemaGenerator` instead? If so, propagate the name change.